src/app/components/post/post.component.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { DateAgoPipe } from 'src/app/pipes/date-ago.pipe';
+import { SubredditFormatPipe } from 'src/app/pipes/subreddit-format.pipe';
+import { PostService } from 'src/app/services/post.service';
+import { Post } from 'src/types/posts';
+import { PostComponent } from './post.component';
+
+class MockPostService {
+    _posts: Post[] = [
+        {
+            id: 1,
+            title: 'First Post',
+            date_created: new Date('2020-11-11T23:28:56.782Z'),
+            subreddit: 'Subreddit1',
+            author: 'Author1',
+            commentCount: 2,
+            post_type: 'text',
+            description: 'Describing this first post.',
+        },
+    ];
+
+    getPost(postId: number): Post {
+        const result = this._posts.find((post) => post.id === postId);
+        return result;
+    }
+}
+
+describe('PostComponent', () => {
+    let component: PostComponent;
+    let fixture: ComponentFixture<PostComponent>;
+    let h2PostTitle: HTMLElement;
+    let divPostDateCreated: HTMLElement;
+    let aPostAuthor: HTMLElement;
+    let aPostSubreddit: HTMLElement;
+    let divPostCommentCount: HTMLElement;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [PostComponent, DateAgoPipe, SubredditFormatPipe],
+            providers: [{ provide: PostService, useClass: MockPostService }],
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(PostComponent);
+        component = fixture.componentInstance;
+        component.postId = 1;
+        h2PostTitle = fixture.nativeElement.querySelector('h2');
+        divPostDateCreated = fixture.nativeElement.querySelector('.postDateCreated');
+        aPostAuthor = fixture.nativeElement.querySelector('.postAuthor');
+        aPostSubreddit = fixture.nativeElement.querySelector('.postSubreddit');
+        divPostCommentCount = fixture.nativeElement.querySelector('.postCommentCount');
+
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('Should display post title', () => {
+        expect(h2PostTitle.textContent).toContain(component.postTitle);
```

Comment:
Doing DOM unit tests is totally fine, but can get quite problematic. Let's leave this here for now and we'll see how far we can get with it.

The main issue is that every test can potentially change the state of your component, and if you don't maintain your tests properly, your DOM tests will more likely than not start failing.

DOM changes are more often done via e2e testing. Nevertheless, this is completely fine for now. I'd suggest to test the functionality of the logic rather than what's shown as it will save you some headaches.